docs/src/contributing/git-workflow.md

```diff
@@ -51,36 +99,173 @@ If the branch exists in a triton/neptune official repo, (as opposed to a persona
+prepare for new features in advance.
+
+It is also an important period of integration testing "in the wild" for the
+release candidate binary.  Operating on a public testnet, it will necessarily be
+exposed to peers running older versions of the software and may shake out issues
+that do not occur with automated testing.
+
+#### Some specifics
+
+1. Each release candidate must be publically announced with a dedicated topic in
+   [talk.neptune.cash](https://talk.neptune.cash).
+2. The release candidate should target 2 weeks of testing after the announcement
+   on average with a minimum of 1 week.
+3. The company behind neptune-cash will provide at least 1 dedicated machine for
+   the public testnet for the purpose of running the release-candidate binary
+   and composing blocks.  Of course the community is also encouraged to run
+   nodes and mine (compose or guess) if possible.
+4. A set of automated tests will be created that utilize the RPC API to perform
+   automated transactions and test basic functionality of the live running
+   nodes. These tests should/must be run at least once against each
+   release-candidate binary while connected to the public testnet network.
+5. If the release-candidate introduces breaking consensus changes, then a
+reset/restart of the testnet from the genesis block may be performed.
+6. Patch releases may skip testnet testing if the risk seems low and/or the
+   urgency is sufficiently high.
 
-The `master` branch can contain changes that are not compatible with whatever network is currently live. Beta-testers looking for the branch that will synchronize with the network that is currently live need branch `release`. This branch may cherry-pick commits that are meant for `master` so long as they are backwards-compatible. However, when this task is too cumbersome, branch `release` will become effectively abandoned -- until the next network version is released.
 
-#### TestNet Release Protocol
+# neptune-core versioning and releases
 
- - Ensure that master builds against crates that live on [crates.io](https://crates.io). In particular, no dependencies on github repositories or revisions.
- - Update `README.md` in case to make sure the installation instructions are up-to-date.
+## deviation from standard semver practice
+
+neptune-core is presently both a (blockchain) library and a binary. Semantic versioning signals changes to APIs and mainly applies to libraries.  But it does not capture breaking changes that can occur in a blockchain ecosystem or breaking changes with regards to stored state, ie databases.
+
+For blockchains it is of primary importance to maintain consensus and network
+compatibility with other nodes on the network.
+
+For these reasons, we define the version identifier as follows.
+
+## version identifier
+
+neptune-core uses sem-ver compatible identifier of the form:
+```
+<major>.<minor>.<patch>[-<candidate>]
```

Comment:
Note that “Cargo uses the convention that only changes in the left-most non-zero component are considered incompatible” ([source](https://doc.rust-lang.org/cargo/reference/semver.html#change-categories)). This means that for releases prior to v1.0.0, the semantic version as interpreted by Cargo are

```
0.0.<major>[-<pre-release>][+<metadata>]
0.<major>.<minor>[-<pre-release>][+<metadata>]
```

Notably, this is different from [semver](semver.org) since there, any two releases prior to v1.0.0 are considered mutually incompatible ([source](https://semver.org/#spec-item-4)).

Do we want to incorporate this in some way?

Reply:
thx.  my feeling is that we really should have changed major version to 1 when we launched mainnet.   Mainnet was a very clear signal to end-users saying that this thing is "ready for production use" (whether true or not) which is also an implied dividing line between 0.x.x and 1.x.x.

Since we didn't, we should still correct it in an upcoming release, probaby at the hard-fork by either going to 1.0.0 or 2.0.0.  Once we do that then any 0.x meanings are irrelevant.  So my feeling is we can ignore that distinction/complication for now.

but if you think differently and would like to suggest wording, I'm open to that.

Reply:
I agree with the “should have been 1.0.0” sentiment. I personally have a weird irrational apprehension to doing a 1.0.0 release. Let's try to get a 1.0.0 going. I agree that the distinction between Cargo's semver and semver.org are irrelevant then, and that a simpler document is favorable.